### PR TITLE
Fix issues caught by golangci-lint

### PR DESCRIPTION
**Pull request by y0ssar1an**
* add missing error checks
* deindent package comments
* remove unnecessary type casts

There are still several more lint issues to fix. This is just a start.

This is branched off #4.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-   Unless required by applicable law or agreed to in writing, software
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
```
Comment (y0ssar1an) [inline cmd/main.go]: `golint` complained about the package comments being indented.

File: pkg/auth/auth.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func authenticate(client *api.Client, config cfg.Config, svc Authenticator) bool
 		log.Printf("could not write token to %s: %v\n", config.TokenPath, err)
 		return false
 	}
-	client.SetToken(string(vaultToken))
+	client.SetToken(vaultToken)
```
Comment (y0ssar1an) [inline pkg/auth/auth.go]: `vaultToken` is already a string.

File: pkg/auth/auth.go [hunk 3]
```diff
@@ -166,13 +166,17 @@ func RenewService(client *api.Client, config cfg.Config) {
 			log.Fatalln("The existing token failed renewal, exiting..")
 		}
 		ttl, err := result.TokenTTL()
+		if err != nil {
+			log.Fatalln("Failed to parse the token's ttl from JSON")
+		}
```
Comment (y0ssar1an) [inline pkg/auth/auth.go]: I'm happy to improve the error the message if you have suggestions.

Comment (broamski) [reply pkg/auth/auth.go]: Works for me!

File: pkg/auth/auth.go (hunk 2 above)
Comment (broamski) [reply pkg/auth/auth.go]: :+1:

File: cmd/main.go (hunk 1 above)
Comment (broamski) [reply cmd/main.go]: :man_shrugging:

**Pull request merged**